README.md

```diff
@@ -54,6 +54,42 @@ GraphQLInterfaceType object = GraphQLAnnotations.iface(SomeInterface.class);
 An instance of the type resolver will be created from the specified class. If a `getInstance` method is present on the
 class, it will be used instead of the default constructor.
 
+## Defining Unions
+
+To have a union, you must annotate an interface with `@GraphQLUnion`. In the annotation, you must declare all the 
+possible types of the union, and a type resolver.
+If no type resolver is specified, `UnionTypeResovler` is used. It follows this algorithm:
+The resolver assumes the the DB entity's name contains the API entity's name, after removing the "Api"
+suffix from the name of the API entity. If so, it takes the result from the dataFetcher and decides to which
+API entity it should be mapped (according to the name). 
+Example: If you have a `Pet` union type, and the dataFetcher returns `DogDB` (or just `Dog`), the typeResolver
+will check for each API entity if its name (without the "Api" suffix - if the suffix exists) is contained 
+inside the name of the DB entity, so both `Dog` and `DogApi` are valid.
+
```

Comment:
>On the other hand in don't understand the example you give in the unit test, where a field of type "Hardware" has a data fetcher which returns an object that is not implementing Hardware

Can you say what test? because I dont rememeber doing such a thing

Reply:
Maybe I missed something, but : the Query type defines a hardwareScreen field of type Hardware - the getHardwareScreen method returns null but the field has a custom dataFetcher "ScreenFetcher" (so method won't be called) - the ScreenFetcher is actually a DataFetcher<ScreenDB>, but ScreenDB does not implement Hardware (in my opinion it should actually throws an error when detecting this - a field should have a DataFetcher of the same type as itself, or compatible, so DataFetcher<Hardware> or DataFetcher<Screen> ). 
Generally, I don't understand what "DB" objects are for, and the resolution from "ScreenDB" to "Screen" (one possible type of the union) is not really obvious .. ?

Reply:
I think that guy ment to create a separation between the GraphQL layer and the BL classes. So while Screen is a class returned by the GraphQL Api layer of the service, the datafetcher is able to return an non-api entity (like ScreenDB, which is used inside the BL or is mapped from the database), and this entity is later mapped to the graphql entity by its fields' datafetchers.

I also believe that this DB/API suffixes is a very private case and is not a standard, so we should not add this behaviour as a default one in the UnionTypeResolver.
I support @tdraier and think that UnionTypeResolver should be matched by name/implementation of the interface, and if someone wants a different behaviour he would implement his own UnionTypeResolver.